pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -2196,8 +2007,8 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 		return
 	}
 	log.Functionf("handleGlobalConfigDelete for %s", key)
-	debug, _ = agentlog.HandleGlobalConfig(log, ctx.subGlobalConfig, agentName,
-		debugOverride, logger)
+	_, _ = agentlog.HandleGlobalConfig(log, ctx.subGlobalConfig, agentName,
```

Comment:
Having HandleGlobalConfig do work to return a debug bool and ignoring it in all the callers makes no sense. Does the current passing of DebugOverride still make sense with this new structure?

Reply:
It seems we can modify `HandleGlobalConfig` function. But it is not direct related to the changes about argument parsing, so let's do it later.

Reply:
But the resulting code in this PR ends up too wierd to read when we have dead code in HandleGlobalConfig to return a debug boolean when all callers ignore it. That ends up being just too confusing. So adding another commit here to remove that debug return makes sense to me.

Reply:
Ah, sorry, I did not mention in my comment in the thread that I removed modification of debug/debug override variables from the PR. Let me check if they will expand the diff too large.

Reply:
Last two commits are about `HandleGlobalConfig` refactoring. Please check, should I move them to the separate PR or keep here.